### PR TITLE
Allow custom media query string in `useMediaQuery`

### PR DESCRIPTION
**Pull request by leroykorterink**

### CODE REVIEW

File: src/hooks/useMediaQuery/useMediaQuery.stories.mdx [hunk 1]
```diff
@@ -4,38 +4,32 @@ import { Meta } from '@storybook/blocks';
 
 # useMediaQuery
 
-Hook that returns a boolean indicating whether the media query matches.
+Hook that returns true when the media query matches.
 
-## Augmenting the MediaQuery type
```
Comment (ThaNarie) [inline src/hooks/useMediaQuery/useMediaQuery.stories.mdx:9]: Is this not possible anymore? I still see an interface defined.

File: src/hooks/useMediaQuery/useMediaQuery.stories.mdx [hunk 2]
```diff
@@ -4,38 +4,32 @@ import { Meta } from '@storybook/blocks';
...
-    '--min-width-1024': never;
-  }
-}
-```
 
 ## Reference
 
 ```ts
-function useMediaQuery(mediaQueryVariableName: keyof MediaQuery, defaultValue = false): boolean;
+function useMediaQuery(mediaQueryOrVariableName: string, defaultValue = false): boolean;
```
Comment (ThaNarie) [inline src/hooks/useMediaQuery/useMediaQuery.stories.mdx:13]: If we type this as `string`, I think we should more clearly document what types of strings you can pass. Having only the 2 examples further down might not clear enough to understand what's really happening.

File: src/hooks/useMediaQuery/useMediaQuery.stories.mdx (hunk 1 above)
Comment (leroykorterink) [reply src/hooks/useMediaQuery/useMediaQuery.stories.mdx:9]: Ah, the docs for the augmenting were incomplete, re-added and improved it.

File: src/hooks/useMediaQuery/useMediaQuery.stories.mdx [hunk 3]
```diff
@@ -4,38 +4,71 @@ import { Meta } from '@storybook/blocks';
...
 ```
+
+### Augmenting the MediaQueryVariables interface
+
+The `MediaQueryVariables` type is used in the type for the `mediaQueryOrVariableName` parameter of
+`useMediaQuery`.
+
+Augment the `MediaQueryVariables` interface to enable auto-complete suggestions for the CSS
+variables that describe media queries in your project. You can still use query strings for the first
+parameter.
```
Comment (ThaNarie) [inline src/hooks/useMediaQuery/useMediaQuery.stories.mdx:58]: > You can still use query strings for the first parameter.

First parameter of what? How does this related to this interface?

Comment (leroykorterink) [reply src/hooks/useMediaQuery/useMediaQuery.stories.mdx:58]: The `useMediaQuery` hook 🙂

**Pull request closed without merging**